File: 014_numeric_cast/main.cpp
#include <cstdint>
#include <limits>
#include <iostream>

int main(void)
{
    int32_t i = 5;
    double d = 1.9;
    uint32_t ui = 10;
    int32_t negative_i = -1;
    uint32_t large_ui = 4294967295;

    std::cout << i << " -> " << static_cast<double>(i) << std::endl;
    std::cout << d << " -> " << static_cast<int32_t>(d) << std::endl;
    std::cout << i << " -> " << static_cast<uint32_t>(i) << std::endl;
    std::cout << ui << " -> " << static_cast<int32_t>(ui) << std::endl;
    std::cout << negative_i << " -> " << static_cast<uint32_t>(negative_i) << std::endl;
    std::cout << large_ui << " -> " << static_cast<int32_t>(large_ui) << std::endl;

    return 0;
}

File: 014_numeric_cast/main.rs
fn main()
{
    let i: i32 = 5;
    let d: f64 = 1.9;
    let ui: u32 = 10;
    let negative_i: i32 = -1;
    let large_ui: u32 = 4294967295;

    println!("{} -> {}", i, i as f64);
    println!("{} -> {}", d, d as i32);
    println!("{} -> {}", i, i as u32);
    println!("{} -> {}", ui, ui as i32);
    println!("{} -> {}", negative_i, negative_i as u32);
    println!("{} -> {}", large_ui, large_ui as i32);
}

